analyzer/src/main/kotlin/managers/SpdxDocumentFile.kt

```diff
@@ -134,21 +135,38 @@ internal fun SpdxExternalDocumentReference.getSpdxPackage(packageId: String, wor
 private fun resolveExternalDocumentReference(
     externalDocumentReference: SpdxExternalDocumentReference,
     workingDir: File
-): File {
+): SpdxDocument {
     val uri = runCatching { URI(externalDocumentReference.spdxDocument) }.getOrElse {
-        throw IllegalArgumentException("'${externalDocumentReference.spdxDocument}' identified by " +
-                "${externalDocumentReference.externalDocumentId} is not a valid URI. }")
+        throw IllegalArgumentException(
+            "'${externalDocumentReference.spdxDocument}' identified by " +
+                    "${externalDocumentReference.externalDocumentId} is not a valid URI. }"
+        )
     }
 
     if (uri.scheme.equals("file", ignoreCase = true) || !uri.isAbsolute) {
         val referencedFile = workingDir.resolve(uri.path)
-        return referencedFile.takeIf { it.isFile }
-            ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file. ")
+        return SpdxModelMapper.read(referencedFile.takeIf { it.isFile }
+            ?: throw IllegalArgumentException("The local file URI '$uri' does not point to an existing file.")
```

Comment:
Suggestion: Extract "referencedFile ... throw IllegalArgumentException" to separate variable.